IS1001Test.py

```diff
@@ -12,12 +12,623 @@
+            return test.FAIL("Scope values MUST be a space-delimited string")
+
+        if not set(RECOMMENDED_RESPONSE_FIELDS).issubset(response.json().keys()):
+            return test.WARNING("Client registration response SHOULD include: {}".format(RECOMMENDED_RESPONSE_FIELDS))
+
+        register_client_schema = self.get_schema(AUTH_API_KEY, "POST", '/register_client', 201)
+
+        try:
+            self.validate_schema(response.json(), register_client_schema)
+            if response.status_code == 201:
+                self.client_data = response.json()
+                return test.PASS()
+        except Exception as e:
+            return test.FAIL("Status code was {} and Schema validation failed. {}".format(response.status_code, e))
+
+        return test.FAIL("Implementation did not pass the necessary criteria.")
+
+    def test_03_token_password_grant(self, test):
+        """Test requesting a Bearer Token using Password Grant from '/token' endpoint"""
+        if self.client_data:
+            for scope in GRANT_SCOPES:
+                request_data = {
+                    'username': AUTH_USERNAME,
+                    'password': AUTH_PASSWORD,
+                    'grant_type': 'password',
+                    'scope': scope
+                }
+                status, response = self._make_auth_request(
+                    "POST", 'token', data=request_data, auth="client"
+                )
+
+                self._verify_response(test=test, expected_status=200, response=response)
+
+                token_schema = self.get_schema(AUTH_API_KEY, "POST", '/token', 200)
+                try:
+                    self.validate_schema(response.json(), token_schema)
+                    if response.status_code == 200:
+                        self.bearer_tokens.append(response.json())
+                        return test.PASS()
+                except Exception as e:
+                    return test.FAIL(
+                        "Status code was {} and Schema validation failed. {}".format(response.status_code, e)
+                    )
+
+                return test.FAIL("Implementation did not pass the necessary criteria.")
+        else:
+            return test.DISABLED("No Client Data available")
+
+    def test_04_authorize_endpoint(self, test):
+        """Test the '/authorize' endpoint and ability to redirect to registered URI with authorization code"""
+
+        if self.client_data:
+            for scope in GRANT_SCOPES:
+
+                state = "xyz"
+                parameters = {
+                    "response_type": "code",
+                    "client_id": self.client_data["client_id"],
+                    "redirect_uri": self.client_data["redirect_uris"][0],
+                    "scope": scope,
+                    "state": state
+                }
+                # NOTE - This is implementation-specific - this signifies the user's consent to the Auth Server

```

Comment:
Is this referring to the line below or above? Any way to remove implementation specific items?